src/main/scala/io/iohk/ethereum/ledger/BlockImport.scala

```diff
@@ -1,364 +1,344 @@
-      newBranch: List[Block],
-      oldBranch: List[BlockData],
-      executedBlocks: List[BlockData]
-  ): Unit = {
-    if (executedBlocks.nonEmpty) {
-      removeBlocksUntil(executedBlocks.head.block.header.parentHash, executedBlocks.last.block.header.number)
-    }
-
-    oldBranch.foreach { case BlockData(block, receipts, weight) =>
-      blockchainWriter.save(block, receipts, weight, saveAsBestBlock = false)
-    }
-
-    import cats.implicits._
-    val checkpointNumber = oldBranch.collect {
-      case BlockData(block, _, _) if block.hasCheckpoint => block.number
-    }.maximumOption
-
-    val bestNumber = oldBranch.last.block.header.number
-    blockchain.saveBestKnownBlocks(bestNumber, checkpointNumber)
-    executedBlocks.foreach(data => blockQueue.enqueueBlock(data.block, bestNumber))
-
-    newBranch.diff(executedBlocks.map(_.block)).headOption.foreach { block =>
-      blockQueue.removeSubtree(block.header.hash)
-    }
-  }
-
-  /** Removes blocks from the [[Blockchain]] along with receipts and total difficulties.
-    *
-    * @param parent     remove blocks until this hash (exclusive)
-    * @param fromNumber start removing from this number (downwards)
-    *
-    * @return the list of removed blocks along with receipts and total difficulties
-    */
-  private def removeBlocksUntil(parent: ByteString, fromNumber: BigInt): List[BlockData] = {
-    @tailrec
-    def removeBlocksUntil(parent: ByteString, fromNumber: BigInt, acc: List[BlockData]): List[BlockData] =
-      blockchainReader.getBestBranch().getBlockByNumber(fromNumber) match {
-        case Some(block) if block.header.hash == parent || fromNumber == 0 =>
-          acc
-
-        case Some(block) =>
-          val hash = block.header.hash
-
-          val blockDataOpt = for {
-            receipts <- blockchainReader.getReceiptsByHash(hash)
-            weight <- blockchain.getChainWeightByHash(hash)
-          } yield BlockData(block, receipts, weight)
-
-          blockchain.removeBlock(hash, withState = true)
-
-          removeBlocksUntil(parent, fromNumber - 1, blockDataOpt.map(_ :: acc).getOrElse(acc))
-
-        case None =>
-          log.error(s"Unexpected missing block number: $fromNumber")
-          acc
-      }
-
-    removeBlocksUntil(parent, fromNumber, Nil)
-  }
-}
-
-sealed trait BlockImportResult
-
-case class BlockImportedToTop(blockImportData: List[BlockData]) extends BlockImportResult
-
-case object BlockEnqueued extends BlockImportResult
-
-case object DuplicateBlock extends BlockImportResult
-
-case class ChainReorganised(
-    oldBranch: List[Block],
-    newBranch: List[Block],
-    weights: List[ChainWeight]
-) extends BlockImportResult
-
-case class BlockImportFailed(error: String) extends BlockImportResult
-
-case class BlockImportFailedDueToMissingNode(reason: MissingNodeException) extends BlockImportResult
-
-case object UnknownParent extends BlockImportResult
+//package io.iohk.ethereum.ledger
```

Comment:
note: if the differ could pick this up as a rename it would be easy to see what changed between the files. as is it treats the file as changed and 'ConsensusImpl' as new.

Reply:
The refactoring of that code is in separate commits